idd/embedded/generate_embeddable_epJSON_schema.cpp

```diff
@@ -71,3 +96,44 @@ namespace EmbeddedEpJSONSchema {
 } // namespace EmbeddedEpJSONSchema
 
 } // namespace EnergyPlus
+)cpp";
+
+int main(int argc, char const* argv[]) {
+  if (argc != 3) {
+    fmt::print(stderr, "usage: ./generate_embeddable_schema path/to/Energy+.schema.epJSON path/to/EmbeddedEpJSONSchema.cc\n");
+    return 1;
+  }
+
+  fmt::print(stderr, "generating the embedded epJSON schema\n");
+
+  std::ifstream schema_stream(argv[1], std::ifstream::in);
+  if (!schema_stream.is_open()) {
+    fmt::print("schema file path {} not found\n", argv[1]);
+    return 1;
+  }
+  auto const input_json = json::parse(schema_stream);
+  std::vector<std::uint8_t> const v_cbor = json::to_cbor(input_json);
+
+  const fs::path outFilePath(argv[2]);
+  const auto outFileDir = outFilePath.parent_path();
+  if (!fs::is_directory(outFileDir)) {
+    fmt::print(stderr, "Output Directory does not exist: {}\n", outFileDir.generic_string());
+    fs::create_directory(outFileDir);
+  }
+
+  auto outfile = fmt::output_file(argv[2], fmt::file::WRONLY | fmt::file::CREATE | fmt::file::TRUNC);
+  outfile.print("{}", header);
+
+  outfile.print("    static constexpr std::array< std::uint8_t, {} > embeddedSchema = {{{{\n", v_cbor.size());
+
+  for (size_t i = 0; i < v_cbor.size(); ++i) {
+    outfile.print("{:#04x},", v_cbor[i]);  // Format the std::uint8_t as hex
+    if (i % 40 == 0 && i != 0) {
+      outfile.print("\n");
+    }
+  }
+  outfile.print("}}}};\n");
+  outfile.print("{}", footer);
+
+  return 0;
+}
```

Comment:
This now takes an extra arg = the path to the output file, and it writes it directly using fmt::output_file.